src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -156715,21 +156715,36 @@ public void UnreachableWhenClause_SwitchExpression()
                 #nullable enable
                 class C
                 {
-                    string? M(bool b, string s)
+                    string? M(bool b, string? s)
                     {
                         return b switch
                         {
-                            _ when s.Length == 0 => s,
+                            _ when s?.Length == 0 => s,
                             _ => null,
                             true when s.Length > 0 => s,
                         };
                     }
                 }
                 """;
-            CreateCompilation(source).VerifyDiagnostics(
+            var comp = CreateCompilation(source).VerifyDiagnostics(
                 // (10,13): error CS8510: The pattern is unreachable. It has already been handled by a previous arm of the switch expression or it is impossible to match.
                 //             true when s.Length > 0 => s,
                 Diagnostic(ErrorCode.ERR_SwitchArmSubsumed, "true").WithLocation(10, 13));
+
+            var syntaxTree = comp.SyntaxTrees.Single();
+            var model = comp.GetSemanticModel(syntaxTree);
+
+            var arms = syntaxTree.GetRoot().DescendantNodes().OfType<SwitchExpressionArmSyntax>().ToImmutableArray();
+            Assert.Equal(3, arms.Length);
+            var unreachableArm = arms[^1];
+
+            var identifiers = unreachableArm.DescendantNodes().OfType<IdentifierNameSyntax>();
```

Comment:
Should we assert anything like the value of `identifiers.Count()` or the string representation of each, to make it more clear what is being tested?